projects/can_transceiver/src/can_transceiver_ros_intf.cpp

```diff
@@ -44,12 +61,120 @@ class CanTransceiverIntf : public rclcpp::Node
+
+        // Publisher
+        publisherWindSensors_ =
+          this->create_publisher<custom_interfaces::msg::WindSensors>(WIND_SENSORS_TOPIC, QUEUE_SIZE);
+        publisherWindSensor_ =
+          this->create_publisher<custom_interfaces::msg::WindSensor>(FILTERED_WIND_SENSOR_TOPIC, QUEUE_SIZE);
+        publisherGPS_ = this->create_publisher<custom_interfaces::msg::GPS>(GPS_TOPIC, QUEUE_SIZE);
+        publisherGenericSensors_ =
+          this->create_publisher<custom_interfaces::msg::GenericSensors>(DATA_SENSORS_TOPIC, QUEUE_SIZE);
+        publisherBatteries_ = this->create_publisher<custom_interfaces::msg::Batteries>(BATTERIES_TOPIC, QUEUE_SIZE);
+        // Timer with 500ms delay
+        timer_ = this->create_wall_timer(500ms, std::bind(&CanSimIntf::timer_callback, this));
+    }
+
+private:
+    // Subscriber
+    void gps_callback(const custom_interfaces::msg::GPS::SharedPtr msg) const {}
+    void wind_callback(const custom_interfaces::msg::WindSensors::SharedPtr msg) const {}
+
+    // Publisher
+    void timer_callback()
+    {
+        // ** GPS **
+        custom_interfaces::msg::GPS gpsPubData;
+        gpsPubData.heading.heading   = PLACEHOLDER_VALUE;
+        gpsPubData.speed.speed       = PLACEHOLDER_VALUE;
+        gpsPubData.lat_lon.latitude  = PLACEHOLDER_VALUE;
+        gpsPubData.lat_lon.longitude = PLACEHOLDER_VALUE;
+        // Publishes msg
+        publisherGPS_->publish(gpsPubData);
+
+        // ** Wind Sensor (WSensor)**
+        custom_interfaces::msg::WindSensor WSensorData;
+        WSensorData.speed.speed = PLACEHOLDER_VALUE;
+        WSensorData.direction   = PLACEHOLDER_VALUE;
+        // Publishes msg
+        publisherWindSensor_->publish(WSensorData);
+
+        // ** Wind Sensors (WSensors)**
+        custom_interfaces::msg::WindSensors WSensorsData;
+        WSensorsData.wind_sensors[0].direction   = PLACEHOLDER_VALUE;
+        WSensorsData.wind_sensors[0].speed.speed = PLACEHOLDER_VALUE;
+        // Publishes msg
+        publisherWindSensors_->publish(WSensorsData);
+
+        // ** Batteries **
+        custom_interfaces::msg::Batteries WBatteriesData;
+        WBatteriesData.batteries[0].voltage = PLACEHOLDER_VALUE;
+        WBatteriesData.batteries[0].current = PLACEHOLDER_VALUE;
+        // Publishes msg
+        publisherBatteries_->publish(WBatteriesData);
+
+        // ** Generic Sensors **
+        custom_interfaces::msg::HelperGenericSensor HelperGenSensorData;
+        custom_interfaces::msg::GenericSensors      GenSensorData;
+        HelperGenSensorData.id   = 0x0;  //uint8
+        HelperGenSensorData.data = 0x0;  //uint64
+        // Publishes msg
+        GenSensorData.generic_sensors.push_back(HelperGenSensorData);
+        publisherGenericSensors_->publish(GenSensorData);
+    }
+
+    // Field Operations
+
+    // Publisher Field Declarations
+    rclcpp::Publisher<custom_interfaces::msg::WindSensors>::SharedPtr    publisherWindSensors_;
```

Comment:
These member field names are waaaaay too similar. Way to easy to use the wrong one with a typo. I'm assuming the singular wind sensor is meant to refer to the filtered_wind_sensor?

Reply:
Updated:
```
    // Field Operations

    // Publisher Field Declarations
    rclcpp::Publisher<custom_interfaces::msg::GPS>::SharedPtr            pub_gps_;
    rclcpp::Publisher<custom_interfaces::msg::WindSensor>::SharedPtr     pub_filtered_wind_sensor_;
    rclcpp::Publisher<custom_interfaces::msg::WindSensors>::SharedPtr    pub_wind_sensors_;
    rclcpp::Publisher<custom_interfaces::msg::Batteries>::SharedPtr      pub_batteries_;
    rclcpp::Publisher<custom_interfaces::msg::GenericSensors>::SharedPtr pub_data_sensors_;

    // Subscriber Field Declarations
    rclcpp::Subscription<custom_interfaces::msg::GPS>::SharedPtr         sub_gps_;
    rclcpp::Subscription<custom_interfaces::msg::WindSensors>::SharedPtr sub_mock_wind_sensors_;

```